refactor: Deprecate the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` (#13063)
shopware/platform: changelog/_unreleased/2025-10-16-deprecate-reflectionhelper-getproperty-getmethod-methods.md

---
title: Deprecate `ReflectionHelper::{getProperty,getMethod}` methods
author: Max
author_email: max@swk-web.com
author_github: @aragon999
---
# Core
* Deprecated `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper::{getProperty,getMethod}` methods
___
# Upgrade Information
## Deprecated `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper::{getProperty,getMethod}` methods

As the accessible modifications are not needed anymore for reflection properties and methods since PHP 8.1, the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper::{getProperty,getMethod}` methods are an overhead, which they should be replaced by their native implementations, i.e. `\ReflectionMethod('My\Class\Name', 'myMethod')` and `\ReflectionProperty('My\Class\Name', 'myMethod')`.
___
# Next Major Version Changes
## Removed `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper::{getProperty,getMethod}` methods

> As the accessible modifications are not needed anymore for reflection properties and methods since PHP 8.1, the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper::{getProperty,getMethod}` methods are an overhead, and therefore are removed. They should be replaced by their native implementations, i.e. `\ReflectionMethod('My\Class\Name', 'myMethod')` and `\ReflectionProperty('My\Class\Name', 'myMethod')`:

Reviewer: ```suggestion
As the accessible modifications are not needed anymore for reflection properties and methods since PHP 8.1, the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper::{getProperty,getMethod}` methods are an overhead, and therefore were removed. They need to be replaced by their native implementations, i.e. `\ReflectionMethod('My\Class\Name', 'myMethod')` and `\ReflectionProperty('My\Class\Name', 'myMethod')`:
```

---
title: Deprecate `ReflectionHelper::{getProperty,getMethod}` methods
author: Max
author_email: max@swk-web.com
author_github: @aragon999
---
# Core
* Deprecated the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`
___
# Upgrade Information
## Deprecated the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`

> As the accessible modifications are not needed anymore for reflection properties and methods since PHP 8.1, the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper::{getProperty,getMethod}` methods are an overhead, which they should be replaced by their native implementations, i.e. `\ReflectionMethod('My\Class\Name', 'myMethod')` and `\ReflectionProperty('My\Class\Name', 'myMethod')`.

Reviewer: I would just write a sentence why the helper is not necessary anymore and then list the old way and what should be used now instead:

asdf bla blu
- case 1
- case 2
- case 3
- case 4

same for the major version section

Author: In the major version section I listed the different replacements, is that sufficient, if so I would simplify the initial sentence and then also list the replacements in the `Upgrade Information` section?

Reviewer: yes, imo that should be enough

Author: Like this?

---

> title: Deprecate `ReflectionHelper::{getProperty,getMethod}` methods

Reviewer: ```suggestion
title: Deprecate `ReflectionHelper`
```

author: Max
author_email: max@swk-web.com
author_github: @aragon999
---
# Core
* Deprecated the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`
___
# Upgrade Information
## Deprecated the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`

> Refection has significantly improved in particular since PHP 8.1, therefore the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` is deprecated, see below for the explicit replacements:

Reviewer: ```suggestion
Refection has significantly improved in particular since PHP 8.1, therefore the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` was deprecated and will be removed in the next major release. See below for the explicit replacements:
```

> author_email: max@swk-web.com
> author_github: @aragon999
> ---
> # Core
> * Deprecated the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`
> ___
> # Upgrade Information
> ## Deprecated the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`
> 
> 
> Refection has significantly improved in particular since PHP 8.1, therefore the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` was deprecated and will be removed in the next major release. See below for the explicit replacements:

Reviewer: ```suggestion
# Core

* Deprecated the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`

___

# Upgrade Information

## Deprecated the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`

Refection has significantly improved in particular since PHP 8.1, therefore the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` was deprecated and will be removed in the next major release. See below for the explicit replacements:
```

```diff
- $property = ReflectionHelper->getProperty(MyClass::class, 'myProperty');
+ $property = \ReflectionProperty(MyClass::class, 'myProperty');
```

```diff
- $method = ReflectionHelper->getMethod(MyClass::class, 'myMethod');
+ $method = \ReflectionMethod(MyClass::class, 'myMethod');
```

```diff
- $propertyValue = ReflectionHelper->getPropertyValue($object, 'myProperty');
+ $propertyValue = \ReflectionProperty(MyClass::class, 'myProperty')->getValue($object);
```

```diff
- $fileName = ReflectionHelper->getFileName(MyClass::class);
+ $fileName = \ReflectionClass(MyClass::class)->getFileName();
```

___
# Next Major Version Changes

> ## Removed the `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`

Reviewer: ```suggestion
## Removal of `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`
```

> Refection has significantly improved in particular since PHP 8.1, therefore the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` is removed, see below for the explicit replacements:

Reviewer: ```suggestion
Refection has significantly improved in particular since PHP 8.1, therefore the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` was removed, see below for the explicit replacements:
```

> ```
> 
> ___
> # Next Major Version Changes
> ## Removal of `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`
> 
> Refection has significantly improved in particular since PHP 8.1, therefore the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` was removed, see below for the explicit replacements:

Reviewer: ```suggestion
___

# Next Major Version Changes

## Removal of `\Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper`

Refection has significantly improved in particular since PHP 8.1, therefore the `Shopware\Core\Framework\Test\TestCaseHelper\ReflectionHelper` was removed, see below for the explicit replacements:
```
